.github/workflows/build.yaml

```diff
@@ -8,9 +8,62 @@
 name: Build FPM Images
 on:
   workflow_dispatch:
+    inputs:
+      version:
+        description: The version to use in the image label
+        required: true
+        type: string
+  push:
+    tags: [ '*' ]
+
 jobs:
-  dummy:
+  build:
+    strategy:
+      matrix:
+        include:
+          - directory: centos/8
+            label: centos8
+          - directory: debian/12
+            label: debian12
     runs-on: ubuntu-latest
+    permissions:
+      id-token: write  # required for aws-actions/configure-aws-credentials
+      contents: read
     steps:
-      - name: Greeting
-        run: echo "This is a dummy build to placate GitHub."
+      - name: version
+        env:
+          VERSION: ${{ inputs.version }}
+          REF_NAME: ${{ github.ref_name }}
+        run: |
+          V="${VERSION:-${REF_NAME}}"
+          if [[ -z "$V" ]]; then
+            echo "No version or ref_name set" >&2
+            exit 1
+          fi
+          echo "V=$V" >> "$GITHUB_ENV"
+      - name: Assume AWS ECR Push Role
+        uses: aws-actions/configure-aws-credentials@010d0da01d0b5a38af31e9c3470dbfdabdecca3a # v4.0.1
+        with:
+          aws-region: us-west-2
+          role-to-assume: arn:aws:iam::146628656107:role/fpm-oci-tag-build-gha
+          role-session-name: "gha-fpm-build=${{ github.run_number }}"
+          role-duration-seconds: 900
+      - name: Log into ECR
+        uses: docker/login-action@v3
```

Comment:
Can we pin this too?  I've been moving everything but `actions/`, `github/`, `gravitational/` and `./` (local repo) references to pinning.